docs/content/specs/terraform/_index.md

```diff
@@ -209,3 +209,361 @@ Module owners **MUST** use the below tooling for unit/linting/static/security an
+  name           = "subnet1"
+  address_prefix = "10.0.1.0/24"
+}
+```
+
+Meta-arguments are assignment statements can be declared by all `resource` or `data` blocks. They are:
+
+* `count`
+* `depends_on`
+* `for_each`
+* `lifecycle`
+* `provider`
+
+The order of declarations within `resource` or `data` blocks is:
+
+All the meta-arguments should be declared on the top of `resource` or `data` blocks in the following order:
+
+1. `provider`
+2. `count`
+3. `for_each`
+
+Then followed by:
+
+1. required arguments
+2. optional arguments
+3. required nested blocks
+4. optional nested blocks
+
+All ranked in alphabetical order.
+
+These meta-arguments should be declared at the bottom of a `resource` block with the following order:
+
+1. `depends_on`
+2. `lifecycle`
+
+The parameters of `lifecycle` block should show up in the following order:
+
+1. `create_before_destroy`
+2. `ignore_changes`
+3. `prevent_destroy`
+
+parameters under `depends_on` and `ignore_changes` are ranked in alphabetical order.
+
+Meta-arguments, arguments and nested blocked are separated by blank lines.
+
+`dynamic` nested blocks are ranked by the name comes after `dynamic`, for example:
+
+```terraform
+  dynamic "linux_profile" {
+    for_each = var.admin_username == null ? [] : ["linux_profile"]
+
+    content {
+      admin_username = var.admin_username
+
+      ssh_key {
+        key_data = replace(coalesce(var.public_ssh_key, tls_private_key.ssh[0].public_key_openssh), "\n", "")
+      }
+    }
+  }
+```
+
+This `dynamic` block will be ranked as a block named `linux_profile`.
+
+Code within a nested block will also be ranked following the rules above.
+
+PS: You can use [`avmfix`](https://github.com/lonegunmanb/azure-verified-module-fix) tool to reformat your code automatically.
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR10 - Category: Code Style - Order within a `module` block
+
+The meta-arguments below should be declared on the top of a `module` block with the following order:
+
+1. `source`
+2. `version`
+3. `count`
+4. `for_each`
+
+blank lines will be used to separate them.
+
+After them will be required arguments, optional arguments, all ranked in alphabetical order.
+
+These meta-arguments below should be declared on the bottom of a `resource` block in the following order:
+
+1. `depends_on`
+2. `providers`
+
+Arguments and meta-arguments should be separated by blank lines.
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR11 - Category: Code Style - Values in `ignore_changes` passed to `provider`, `depends_on`, `lifecycle` blocks are not allowed to use double quotations
+
+Good example:
+
+```hcl
+lifecycle {
+    ignore_changes = [
+      tags,
+    ]
+}
+```
+
+Bad example:
+
+```hcl
+lifecycle {
+    ignore_changes = [
+      "tags",
+    ]
+}
+```
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR12 - Category: Code Style - For resources have configurable `tags` field, `tags` should be always exposed to module users through `variable` to ensure they are able to set `tags`
```

Comment:
Is this not a duplicate of [RMFR4](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-rmfr4---category-composition---avm-consistent-feature--extension-resources-value-add)?

If so lets remove it to keep only a single copy of the NFR

Reply:
Good catch! Thanks @jtracey93 !